server/routes/pilots/stripe.js

```diff
@@ -20,81 +20,93 @@ function pilotRequired(req, res, next) {
+      accountId = account.id;
+
+      // Update the model and store the Stripe account ID in the datastore:
+      // this Stripe account ID will be used to issue payouts to the pilot
+      req.user.stripeAccountId = accountId;
+      await req.user.save();
+    }
+
+    // Create an account link for the user's Stripe account
+    const accountLink = await stripe.accountLinks.create({
+      account: accountId,
+      refresh_url: config.publicDomain + '/pilots/stripe/authorize',
+      return_url: config.publicDomain + '/pilots/stripe/onboarded',
+      type: 'account_onboarding'
+    });
+
+    // Redirect to Stripe to start the Express onboarding flow
+    res.redirect(accountLink.url);
+  } catch (err) {
+    console.log('Failed to create a Stripe account.');
+    console.log(err);
+    res.redirect('/pilots/signup');
+  }
 });
 
 /**
- * GET /pilots/stripe/token
+ * GET /pilots/stripe/onboarded
  *
- * Connect the new Stripe account to the platform account.
+ * Return endpoint from Stripe onboarding, checks if onboarding has been completed
  */
-router.get('/token', pilotRequired, async (req, res, next) => {
-  // Check the `state` we got back equals the one we generated before proceeding (to protect from CSRF)
-  if (req.session.state != req.query.state) {
-    return res.redirect('/pilots/signup');
-  }
+router.get('/onboarded', pilotRequired, async (req, res) => {
   try {
-    // Post the authorization code to Stripe to complete the Express onboarding flow
-    const expressAuthorized = await request.post({
-      uri: config.stripe.tokenUri, 
-      form: { 
-        grant_type: 'authorization_code',
-        client_id: config.stripe.clientId,
-        client_secret: config.stripe.secretKey,
-        code: req.query.code
-      },
-      json: true
-    });
+    // Retrieve the user's Stripe account and check if they have finished onboarding
+    const account = await stripe.account.retrieve(req.user.stripeAccountId);
+    if (account.details_submitted) {
+      req.user.onboardingComplete = true;
+      await req.user.save();
 
-    if (expressAuthorized.error) {
-      throw(expressAuthorized.error);
+      // Redirect to the Rocket Rides dashboard
+      req.flash('showBanner', 'true');
+      res.redirect('/pilots/dashboard');
+    } else {
+      console.log('The onboarding process was not completed.');
+      res.redirect('/pilots/signup');
     }
-
-    // Update the model and store the Stripe account ID in the datastore:
-    // this Stripe account ID will be used to issue payouts to the pilot
-    req.user.stripeAccountId = expressAuthorized.stripe_user_id;
-    await req.user.save();
-
-    // Redirect to the Rocket Rides dashboard
-    req.flash('showBanner', 'true');
-    res.redirect('/pilots/dashboard');
   } catch (err) {
-    console.log('The Stripe onboarding process has not succeeded.');
-    next(err);
+    console.log(err);
+    res.redirect('/pilots/signup');
```

Comment:
It looks an error in the previous version of this endpoint would result in an error page like this:

![image](https://user-images.githubusercontent.com/89594290/184431437-4ecd39a1-3d69-49ce-adcb-2e505c6f2d91.png)

Which is kind of nice for demo purposes. The new version silently swallows the error and redirects back to the signup page. I think I slightly prefer the old behavior (although I don't feel too strongly about it). WDYT?

Reply:
Sure, I think that makes sense to me. Surfacing the error is probably more clear to the user than silently redirecting.